arcgis-ios-sdk-samples/Display information/Display grid/DisplayGridSettingsViewController.swift

```diff
@@ -241,77 +241,105 @@ class DisplayGridSettingsViewController: UITableViewController {
-        case labelFormatCell:
-            if let selectedIndex = (grid as? AGSLatitudeLongitudeGrid)?.labelFormat.rawValue {
-                let optionsViewController = OptionsTableViewController(labels: labelFormatLabels, selectedIndex: selectedIndex) { (newIndex) in
-                    if let grid = self.mapView?.grid as? AGSLatitudeLongitudeGrid {
-                        grid.labelFormat = AGSLatitudeLongitudeGridLabelFormat(rawValue: newIndex)!
-                        self.updateLabelFormatUI()
-                    }
-                }
-                optionsViewController.title = "Format"
-                show(optionsViewController, sender: self)
-            }
-        case labelUnitCell:
-            if let selectedIndex = (grid as? AGSUSNGGrid)?.labelUnit.rawValue ?? (grid as? AGSMGRSGrid)?.labelUnit.rawValue {
-                let optionsViewController = OptionsTableViewController(labels: labelUnitLabels, selectedIndex: selectedIndex) { (newIndex) in
-                    if let grid = self.mapView?.grid as? AGSMGRSGrid {
-                        grid.labelUnit = AGSMGRSGridLabelUnit(rawValue: newIndex)!
-                    } else if let grid = self.mapView?.grid as? AGSUSNGGrid {
-                        grid.labelUnit = AGSUSNGGridLabelUnit(rawValue: newIndex)!
-                    }
-                    self.updateLabelUnitUI()
+        }
+    }
+    
+    private func showLabelUnitPicker() {
+        if let grid = mapView?.grid,
+            let selectedIndex = (grid as? AGSUSNGGrid)?.labelUnit.rawValue ?? (grid as? AGSMGRSGrid)?.labelUnit.rawValue {
+            let optionsViewController = OptionsTableViewController(labels: labelUnitLabels, selectedIndex: selectedIndex) { (newIndex) in
+                if let grid = self.mapView?.grid as? AGSMGRSGrid {
+                    grid.labelUnit = AGSMGRSGridLabelUnit(rawValue: newIndex)!
+                } else if let grid = self.mapView?.grid as? AGSUSNGGrid {
+                    grid.labelUnit = AGSUSNGGridLabelUnit(rawValue: newIndex)!
                 }
-                optionsViewController.title = "Unit"
-                show(optionsViewController, sender: self)
-            }
-        case gridColorCell:
-            guard let color = gridColor(of: grid) else {
-                return
-            }
-            let controller = ColorPickerViewController.instantiateWith(color: color) { (color) in
-                self.changeGridColor(of: grid, to: color)
-                self.updateUIForGridColor()
+                self.updateLabelUnitUI()
             }
-            controller.title = "Grid Color"
-            show(controller, sender: self)
-        case labelColorCell:
-            guard let color = labelColor(of: grid) else {
-                return
+            optionsViewController.title = "Unit"
+            show(optionsViewController, sender: self)
+        }
+    }
+    
+    private func showGridTypePicker() {
+        if let grid = mapView?.grid,
```

Comment:
Would this make more sense as a guard statement?

Reply:
👍